### PR TITLE
Fix up rtweekend.h introduction and use

### LINKED ISSUE #1473: Fix up rtweekend.h introduction
In book 1, once we introduce `rtweekend.h`, we need to include the changes to all other existing header files to include it, and to remove now-unneeded declarations and definitions.

### PR DESCRIPTION
**Pull request by hollasch**
Resolves #1473

### CODE REVIEW

File: books/RayTracingInOneWeekend.html [hunk 1]
```diff
@@ -2884,6 +2936,15 @@
 In service of this, we'll create a new vector method -- `vec3::near_zero()` -- that returns true if
 the vector is very close to zero in all dimensions.
 
+We'll need to use the C++ standard library function `std::fabs`, which returns the absolute value of
+its input. We'll add this to `rtweekend.h` since we'll need this in several locations as we proceed.
+
+
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++
+    XXXX
```
Comment (armansito) [inline books/RayTracingInOneWeekend.html]: Missing code snippet?

File: books/RayTracingInOneWeekend.html [hunk 2]
```diff
@@ -1391,19 +1391,20 @@
...
-file.
+also throw common useful constants and future utility functions in `rtweekend.h`, our general main
+header file.
 
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++
     #ifndef RTWEEKEND_H
     #define RTWEEKEND_H
 
     #include <cmath>
+    #include <iostream>
```
Comment (armansito) [inline books/RayTracingInOneWeekend.html:1402]: Should there be a `#include <cstdlib>` here to match the code?

File: books/RayTracingInOneWeekend.html [hunk 3]
```diff
@@ -1429,8 +1430,56 @@
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     [Listing [rtweekend-initial]: <kbd>[rtweekend.h]</kbd> The rtweekend.h common header]
 
+For all main program files, we will include `rtweekend.h` first, so most other header files (where
+the bulk of our code will reside) can assume that they are free to assume that these definitions are
```
Comment (armansito) [inline books/RayTracingInOneWeekend.html]: > ...can assume that they are free to assume...

..._can assume that these definitions are already available._ ...

File: books/RayTracingInOneWeekend.html [hunk 4]
```diff
@@ -1429,8 +1430,56 @@
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     [Listing [rtweekend-initial]: <kbd>[rtweekend.h]</kbd> The rtweekend.h common header]
 
+For all main program files, we will include `rtweekend.h` first, so most other header files (where
```
Comment (armansito) [inline books/RayTracingInOneWeekend.html]: I would rephrase this to:

_All main program files will include `rtweekend.h` first, so most..._

File: books/RayTracingInOneWeekend.html [hunk 5]
```diff
@@ -1429,8 +1430,56 @@
...
+the bulk of our code will reside) can assume that they are free to assume that these definitions are
+already available. (Headers included inside `rtweekend.h` still need to include any of their
+dependencies.) Because of this, we'll make some updates with this assumption in mind.
```
Comment (armansito) [inline books/RayTracingInOneWeekend.html]: I would drop the "Because of this, ".

**Review by armansito — COMMENTED** [review]
Just a few suggestions. Otherwise looks good.

File: books/RayTracingInOneWeekend.html (hunk 1 above)
Comment (hollasch) [reply books/RayTracingInOneWeekend.html]: Oops. Other work in progress that slipped in here.

File: books/RayTracingInOneWeekend.html (hunk 2 above)
Comment (hollasch) [reply books/RayTracingInOneWeekend.html:1402]: Will fix.

File: books/RayTracingInOneWeekend.html (hunk 1 above)
Comment (hollasch) [reply books/RayTracingInOneWeekend.html]: Actually, I'll bring this forward to resolve here.

File: books/RayTracingInOneWeekend.html (hunk 2 above)
Comment (hollasch) [reply books/RayTracingInOneWeekend.html:1402]: Nope, not needed yet.

Comment (hollasch) [reply books/RayTracingInOneWeekend.html:1402]: See line 2002.

File: books/RayTracingInOneWeekend.html (hunk 4 above)
Comment (hollasch) [reply books/RayTracingInOneWeekend.html]: Done.

File: books/RayTracingInOneWeekend.html (hunk 3 above)
Comment (hollasch) [reply books/RayTracingInOneWeekend.html]: Done.

File: books/RayTracingInOneWeekend.html (hunk 5 above)
Comment (hollasch) [reply books/RayTracingInOneWeekend.html]: Done. Man, I really had a mouth full of marbles on this one.

File: books/RayTracingInOneWeekend.html (hunk 1 above)
Comment (hollasch) [reply books/RayTracingInOneWeekend.html]: Done.

File: books/RayTracingInOneWeekend.html [hunk 6]
```diff
@@ -1391,19 +1391,20 @@
 We need some math constants that we conveniently define in their own header file. For now we only
 need infinity, but we will also throw our own definition of pi in there, which we will need later.
 There is no standard portable definition of pi, so we just define our own constant for it. We'll
-throw common useful constants and future utility functions in `rtweekend.h`, our general main header
-file.
+also throw common useful constants and future utility functions in `rtweekend.h`, our general main
+header file.
```
Comment (rupsis) [inline books/RayTracingInOneWeekend.html:1395]: Nit: This is a bit wordy

```
We'll also throw useful common constants, and future utility functions in `rtweekend.h`. 
This will be our general main header file.
```

File: books/RayTracingInOneWeekend.html [hunk 7]
```diff
@@ -1429,8 +1430,56 @@
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     [Listing [rtweekend-initial]: <kbd>[rtweekend.h]</kbd> The rtweekend.h common header]
 
+All main program files will include `rtweekend.h` first, so most other header files (where the bulk
+of our code will reside) can assume that these definitions are already available. (Headers included
```
Comment (rupsis) [inline books/RayTracingInOneWeekend.html:1434]: ~that~

File: books/RayTracingInOneWeekend.html [hunk 8]
```diff
@@ -1429,8 +1430,56 @@
...
+inside `rtweekend.h` still need to include any of their dependencies.) We'll make some updates with
```
Comment (rupsis) [inline books/RayTracingInOneWeekend.html:1435]: Starting a sentence with a parenthetical feels a bit odd.  I think we can just omit the parenthesis all together an have it be a standalone sentence.

File: books/RayTracingInOneWeekend.html [hunk 9]
```diff
@@ -1429,8 +1430,56 @@
...
+this assumption in mind.
+
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++ delete
+    #include "vec3.h"
+
+    #include <iostream>
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    [Listing [assume-rtw-color]: <kbd>[color.h]</kbd> Assume rtweekend.h in color.h]
```
Comment (rupsis) [inline books/RayTracingInOneWeekend.html:1443]: Don't love the term "Assume".  

`Cleanup x.h in favor of rtweekend.h` might read a little better.

File: books/RayTracingInOneWeekend.html [hunk 10]
```diff
@@ -2884,6 +2936,23 @@
...
+
+
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++ highlight
+    using std::fabs;
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++
+    using std::make_shared;
+    using std::shared_ptr;
+    using std::sqrt;
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    [Listing [declare-fabs]: <kbd>rtweekend.h</kbd> Declaring std::fabs()
```
Comment (rupsis) [inline books/RayTracingInOneWeekend.html:2954]: Missing closing `]` 

image

File: books/RayTracingInOneWeekend.html [hunk 11]
```diff
@@ -2884,6 +2936,23 @@
 In service of this, we'll create a new vector method -- `vec3::near_zero()` -- that returns true if
 the vector is very close to zero in all dimensions.
 
+We'll need to use the C++ standard library function `std::fabs`, which returns the absolute value of
+its input. We'll add this to `rtweekend.h` since we'll use this in several future locations.
+
```
Comment (rupsis) [inline books/RayTracingInOneWeekend.html:2941]:
```
We'll add this to `rtweekend.h` since it'll be used in several future locations.
```
Multiple use of `we'll` is a bit awkward.

**Review by rupsis — CHANGES_REQUESTED** [review]
Mostly editing suggestions. 

Only requesting changes because missing `]` breaks page rendering.

File: books/RayTracingInOneWeekend.html (hunk 10 above)
Comment (hollasch) [reply books/RayTracingInOneWeekend.html:2954]: Fixed in second commit.

Comment (hollasch) [reply books/RayTracingInOneWeekend.html:2954]: Oops — no it's not.

File: books/RayTracingInOneWeekend.html (hunk 9 above)
Comment (hollasch) [reply books/RayTracingInOneWeekend.html:1443]: "in favor of" doesn't make sense to me. I can change to "Assume rtweekend.h inclusion for".

File: books/RayTracingInOneWeekend.html (hunk 2 above)
Comment (armansito) [reply books/RayTracingInOneWeekend.html:1402]: Ah, yeah I missed its introduction down the line.

**Review by armansito — COMMENTED** [review]
Belated approval
